### ISSUE TITLE
minion/server.go: should not attempt to kill a process that's already dead

### ISSUE THREAD

**Issue by jdef** [mesosphere/kubernetes-mesos#453]
from the `stderr` logs on DCOS running v1.0.3-v0.6.3 (ignore the formatting error, it's already patched):
```
I0820 20:33:42.434524    3895 server.go:171] killing proxy process...
E0820 20:33:42.434530    3895 server.go:217] %v exited with error: %vproxyexit status 255
E0820 20:33:42.434541    3895 server.go:173] failed to kill proxy process: os: process already finished
I0820 20:33:47.466592    3895 server.go:204] wrote 203 bytes to proxy.log
I0820 20:33:47.466621    3895 server.go:171] killing proxy process...
E0820 20:33:47.466631    3895 server.go:173] failed to kill proxy process: os: process already finished
E0820 20:33:47.466636    3895 server.go:217] %v exited with error: %vproxyexit status 255
I0820 20:33:52.499907    3895 server.go:204] wrote 203 bytes to proxy.log
I0820 20:33:52.499943    3895 server.go:171] killing proxy process...
E0820 20:33:52.499944    3895 server.go:217] %v exited with error: %vproxyexit status 255
E0820 20:33:52.499955    3895 server.go:173] failed to kill proxy process: os: process already finished
```

**Comment by sttts** [comment]
I am not even sure we kill the processes correctly when needed. Will check.

**Comment by sttts** [comment]
If somebody wants to approach this while I am on vacation:

- The relevant code is `launchHyperkubeServer` in server.go.
- The current code os.Waits for each process (kube-proxy and executor). When the executor terminates cleanly (return code 0), the proxy should be killed as well. But the go routine of the proxy keeps waiting. This must be changed I guess.
- Each sub-process has it log-write go routine running a `IO.Copy`. When this go-routine quits, the process is killed. There should be a signal 0 sent to the process first, checking the error. Compare http://stackoverflow.com/questions/15204162/check-if-a-process-exists-in-go-way

**Comment by jdef** [comment]
Yes this should be fixed for the k8s/dcos round1 refresh

**Comment by jdef** [comment]
fixed by #458
